pennylane/math/matrix_manipulation.py

```diff
@@ -113,9 +113,12 @@ def expand_matrix(mat, wires, wire_order=None, sparse_format="csr"):
     if (wire_order is None) or (wire_order == wires):
         return mat
 
-    if not wires and qml.math.shape(mat) == (qudit_dim, qudit_dim):
+    if not wires:
         # global phase
-        wires = wire_order[0:1]
+        if qml.math.shape(mat) == (qudit_dim, qudit_dim):
+            wires = wire_order[0:1]
```

Comment:
It seems that the line119 has never been visited by the whole codebase. This means it actually only serves for the previous wrong impelmentation of `GlobalPhase(.., wires=None)`. So we will just delete this